src/cloud-sql-instance.ts

```diff
@@ -75,7 +76,25 @@ export class CloudSQLInstance {
     this.sqlAdminFetcher = sqlAdminFetcher;
   }
 
+  async forceRefresh(): Promise<void> {
+    // if a refresh is already ongoing, just await for its promise to fulfill
+    // so that a new instance info is available before reconnecting
+    if (this.ongoingRefreshPromise) {
+      await this.ongoingRefreshPromise;
+      return;
+    }
+    this.close();
+    this.closed = false;
```

Comment:
IMHO looks confusing to read this code. First it is closed, then on the very next line it is not closed.

I suggest refactoring out the piece of code which clears the timer, and call it from both `forceRefresh` and `close`.

Reply:
Landed and rebased https://github.com/GoogleCloudPlatform/cloud-sql-nodejs-connector/pull/198 - this is definitely more clear now, thanks for the help!